packages/client/src/components/domain/shop/ManufacturerProductListPage.tsx

```diff
@@ -0,0 +1,92 @@
+import { Column, Table } from '@/components/case/Table';
+import { useEffect, useState } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import * as shopApi from '@/api/shop';
+import { useAuthLoaderData } from '@/hooks/useAuthLoaderData';
+import { toast } from 'react-hot-toast';
+
+type FetchHandlingProductsResponse = Awaited<ReturnType<typeof shopApi.fetchHandlingProducts>>;
+
+const useHandleProducts = (manufacturerId: string) => {
+  const loaderData = useAuthLoaderData();
+  const shopId = loaderData.id;
+  const token = loaderData.token;
+
+  const [products, setProducts] = useState<FetchHandlingProductsResponse | null>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    shopApi
+      .fetchHandlingProducts({ shopId, manufacturerId, token })
+      .then((products) => {
+        setProducts(products);
+        setError(null);
+      })
+      .catch((error: Error) => {
+        setError(error);
+        setProducts(null);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [shopId, manufacturerId, token]);
```

Comment:
[IMO] stateにレスポンスを丸ごと格納するのではなく、productsの配列データのみを持たせるようにした方がコンポーネント側でのコード量を減らせそうです

```suggestion
  const [products, setProducts] = useState<FetchHandlingProductsResponse['products']>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState<Error | null>(null);

  useEffect(() => {
    setIsLoading(true);
    shopApi
      .fetchHandlingProducts({ shopId, manufacturerId, token })
      .then((res) => {
        const products = res.products;
        setProducts(products);
        setError(null);
      })
      .catch((error: Error) => {
        setError(error);
        setProducts(null);
      })
      .finally(() => {
        setIsLoading(false);
      });
  }, [shopId, manufacturerId, token]);
```